TypeCobol/Compiler/Scanner/Token.cs

```diff
@@ -297,11 +302,22 @@ public virtual string Text
         /// </summary>
         public LiteralTokenValue LiteralValue { get; set; }
 
+        private MultilineScanState scanStateSnapshot;
         /// <summary>
         /// ScanState associated to this token if any, null otherwise.
         /// This property is used to allow PartialCobolWords proper reconstruction.
         /// </summary>
-        public MultilineScanState ScanStateSnapshot { get; set; }
+        public MultilineScanState ScanStateSnapshot
+        {
+            get
+            {
+                return scanStateSnapshot ?? (this.TokensLine is TokensLine tl ? tl.ScanState : null);
+            }
+            private set
```

Comment:
You can remove this setter, make the field `scanStateSnapshot` readonly and set the field value (instead of calling the setter) in constructor.

Reply:
So this means that this value must be always initialized to (this.TokensLine is TokensLine tl ? tl.ScanState : null); by default.
ok

Reply:
No this does not change the initial value. Keep the property getter and replace the setter call with field assignment. The initial value and logic stays the same.

Reply:
ok done in 5e55727